### ISSUE TITLE
Nuke metal_SocDescriptor

### ISSUE THREAD

**Issue by broskoTT** [tenstorrent/tt-umd#331]
Analyze what features are missing in our own umd soc_descriptor, and remove metal_socdescriptor class altogether from tt_metal.

**Comment by broskoTT** [comment]
To be done after #243 is done. And after tt_metal is refactored not to fetch any harvesting info from the tt::umd

**Comment by broskoTT** [comment]
TBD after #439

**Comment by broskoTT** [comment]
Not a priority and not sure if it is possible.
dram_view is read by metal_soc_descriptor, so it still has something to hold...

**Comment by broskoTT** [comment]
Should be renamed and contain only preferred view for DRAM, and not wrap our soc descriptor
